pkg/lib/bundle/generate.go

```diff
@@ -320,22 +314,35 @@ func ValidateChannelDefault(channels, channelDefault string) (string, error) {
 	return chanDefault, chanErr
 }
 
+type Annotations struct {
+	MediaType        string
+	ManifestsDirName string
+	MetadataDirName  string
+	PackageName      string
+	Channels         string
+	ChannelDefault   string
+}
+
+func (a *Annotations) IsComplete() bool {
+	return a.ManifestsDirName != "" && a.MetadataDirName != "" && a.Channels != "" && a.PackageName != "" && a.MediaType != ""
+}
+
 // GenerateAnnotations builds annotations.yaml with mediatype, manifests &
 // metadata directories in bundle image, package name, channels and default
 // channels information.
-func GenerateAnnotations(mediaType, manifests, metadata, packageName, channels, channelDefault string) ([]byte, error) {
+func GenerateAnnotations(a Annotations) ([]byte, error) {
 	annotations := &AnnotationMetadata{
 		Annotations: map[string]string{
```

Comment:
If we are defining an explicit type for these annotations, why use a map here? Is it just to ensure that output isn't printed? Could we just use json output tags for that?

Reply:
Well, i changed this to a struct so that we don't pass around 6 things in and out of a func